### PR TITLE
BaSD-MS-28: Update profiles logic

### PR DESCRIPTION
**Pull request by leimeter-joaquin**
# Profiles:

| Property  | Type | Required | 
| ------------- | ------------- | ------------- |
| name | String | ✔️ |

### Methods:

- listProfiles
- createProfile
- updateProfile
- deleteProfile

### CODE REVIEW

File: controllers/profiles.js [hunk 1]
```diff
@@ -1,120 +1,53 @@
...
-  }
-
-  return fs.writeFile('./data/profiles.json', JSON.stringify(profilesUpdated), (err) => {
-    if (err) {
-      return errorResHelper(err, res);
+    if (error) {
+      return res.status(400).json(error);
     }
-    return res.status(204).json(removedProfile);
+    return res.status(204).send();
```
Comment (estefrare) [inline controllers/profiles.js:49]: send a message

File: controllers/profiles.js [hunk 2]
```diff
@@ -1,120 +1,53 @@
...
   });
 };
 
 const listProfiles = (req, res) => {
-  if (!profiles.length) {
-    return errorResHelper('The Profiles List seems to be empty.', res, 404);
-  }
-  return res.status(200).json(profiles);
+  Profiles.find()
```
Comment (estefrare) [inline controllers/profiles.js]: add filters

File: controllers/profiles.js [hunk 3]
```diff
@@ -1,120 +1,53 @@
...
+    .then((profiles) => res.status(200).json(profiles))
+    .catch((error) => res.status(400).json(error));
```
Comment (estefrare) [inline controllers/profiles.js]: update the format error

File: validations/profiles.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  return next();
+};
+
+const validateIdFormat = (req, res, next) => {
+  if (!ObjectId.isValid(req.params.id)) {
+    return res.status(400).send('Profile id is not valid');
+  }
+  return next();
+};
+
```
Comment (estefrare) [inline validations/profiles.js:18]: you should validate the name format

**Review by estefrare — CHANGES_REQUESTED** [review]

**Comment by estefrare** [comment]
@leimeterjoaquin Please, remove the `profiles.json`

File: controllers/profiles.js (hunk 1 above)
Comment (leimeter-joaquin) [reply controllers/profiles.js:49]: But when status is 204 no message is sent, right?

Comment (estefrare) [reply controllers/profiles.js:49]: you are right! 204 should not response with data

File: controllers/profiles.js [hunk 5]
```diff
@@ -1,122 +1,55 @@
-const fs = require('fs');
-const profiles = require('../data/profiles.json');
+const Profiles = require('../models/Profiles');
 
-const errorResHelper = (errorDescription, res, errCode = 400) => {
-  res.status(errCode).json({ error: errorDescription });
+const listProfiles = (req, res) => {
+  Profiles.find(req.query)
+    .then((profiles) => res.status(200).json(profiles))
+    .catch((error) => res.status(400).json(error));
```
Comment (ignacioaristo) [inline controllers/profiles.js]: Add the `{message: error}`

File: controllers/profiles.js [hunk 6]
```diff
@@ -1,122 +1,55 @@
...
+  const profile = new Profiles({
+    name: req.body.name,
+  });
 
-  return fs.writeFile('./data/profiles.json', JSON.stringify(profiles), (err) => {
-    if (err) {
-      return errorResHelper(err, res);
+  profile.save((error) => {
+    if (error) {
+      return res.status(400).json(error);
```
Comment (ignacioaristo) [inline controllers/profiles.js]: add the "message"

File: controllers/profiles.js [hunk 7]
```diff
@@ -1,122 +1,55 @@
...
+    {
+      name: req.body.name,
+    },
+    { new: true },
+    (error, newProfile) => {
+      if (!newProfile) {
+        return res.status(404).send({ Message: 'Profile id does not exist' });
+      }
+      if (error) {
+        return res.status(400).json(error);
```
Comment (ignacioaristo) [inline controllers/profiles.js]: Same here

File: controllers/profiles.js [hunk 8]
```diff
@@ -1,122 +1,55 @@
...
-      res,
-      404,
-    );
-  }
-
-  return fs.writeFile('./data/profiles.json', JSON.stringify(profilesUpdated), (err) => {
-    if (err) {
-      return errorResHelper(err, res);
+    if (error) {
+      return res.status(400).json(error);
```
Comment (ignacioaristo) [inline controllers/profiles.js]: Same here

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: postman-collection/BaSD - MindSet Postman Collection.postman_collection.json [hunk 9]
```diff
@@ -59,14 +59,11 @@
 						"method": "GET",
 						"header": [],
 						"url": {
-							"raw": "http://localhost:4000/api/profiles/",
-							"protocol": "http",
+							"raw": "{{url}}/profiles/",
```
Comment (estefrare) [inline postman-collection/BaSD - MindSet Postman Collection.postman_collection.json]: we decided not to use environment in Postman yet

**Review by estefrare — CHANGES_REQUESTED** [review]

**Review by NicoCagnina — APPROVED** [review]

**Review by estefrare — APPROVED** [review]

**Review by estefrare — APPROVED** [review]

**Review by ignacioaristo — APPROVED** [review]

**Pull request merged**